src/hooks/useDeleteOrganization.ts

```diff
@@ -0,0 +1,69 @@
+import { useState } from "react";
+
+import omit from "lodash/omit";
+import { useTranslation } from "react-i18next";
+import { useNavigate } from "react-router-dom";
+
+import { LoggerService, ProjectsService } from "@services";
+import { namespaces } from "@src/constants";
+import { ModalName } from "@src/enums/components";
+import { useModalStore, useOrganizationStore } from "@src/store";
+import { EnrichedOrganization } from "@src/types/models";
+
+export const useDeleteOrganization = () => {
+	const [organizationIdInDeletion, setOrganizationIdInDeletion] = useState<string>();
+	const { closeModal } = useModalStore();
+	const { currentOrganization, user, deleteOrganization, logoutFunction } = useOrganizationStore();
+	const { t } = useTranslation("settings", { keyPrefix: "userOrganizations" });
+	const navigate = useNavigate();
+
+	const onDelete = async (organization: EnrichedOrganization) => {
+		const deletingCurrentOrganization = organization.id === currentOrganization?.id;
+
+		const { error } = await deleteOrganization(omit(organization, "currentMember"));
+		closeModal(ModalName.deleteOrganization);
+		if (error) {
+			return { error: true };
+		}
+
+		if (!deletingCurrentOrganization) return { error: false };
+		setTimeout(async () => {
+			if (!user?.defaultOrganizationId) {
+				LoggerService.error(
+					namespaces.ui.organizationTableUserSettings,
+					t("errors.defaultOrganizationIdMissing", { userId: user?.id })
+				);
+				logoutFunction(true);
+				return;
+			}
+			navigate(`/switch-organization/${user.defaultOrganizationId}`);
+		}, 3000);
+		return { error: false };
+	};
+
+	const handleDeleteOrganization = async (organization: EnrichedOrganization) => {
+		setOrganizationIdInDeletion(organization.id);
+
+		try {
```

Comment:
I don't think we need `try-catch` here because `ProjectsService.list(organization.id)` is already wrapped with `try-catch` and I don't see other code that can fail here.

Reply:
But in which scenario can you get an exception here? from which code?